### PR TITLE
service/paxos/paxos_state: restore resilience against dropped tables

### PR DESCRIPTION
**Pull request by denesb**
Recently, the code in paxos_state::prepare(), paxos_state::accept() and paxos_state::learn() was coroutinized by 58912c2cc1c, 887a5a8f625 and 2b7acdb32c6 respectively. This introduced a regression: the latency histogram updater code, was moved from a finally() to a defer(). Unlike the former, the latter runs in a noexcept context so the possible replica::no_such_column_family raised from the latency update code now crashes the node, instead of failing just the paxos operation as before. Fix by only updating the latency histogram if the table still exists.

Fixes: scylladb/scylladb#19620

No vulnerable versions, no backport needed.

### CODE REVIEW

**Review by kbr-scylla — APPROVED** [review]
Thanks for the quick fix.

File: service/paxos/paxos_state.cc [hunk 1]
```diff
@@ -60,8 +60,12 @@ future<prepare_response> paxos_state::prepare(storage_proxy& sp, db::system_keys
     lc.start();
 
     auto stats_updater = defer([&sp, schema, lc] () mutable {
-        auto& stats = sp.get_db().local().find_column_family(schema).get_stats();
-        stats.cas_prepare.mark(lc.stop().latency());
+        try {
```
Comment (bhalevy) [inline service/paxos/paxos_state.cc:63]: How about the following instead:
```c++
    if (auto table = sp.get_db().local().get_tables_metadata().get_table_if_exists(schema->id())) {
            auto& stats = table->get_stats();
            stats.cas_prepare.mark(lc.stop().latency());
    }
```

Comment (denesb) [reply service/paxos/paxos_state.cc:63]: I was looking for this method, but didn't find it.

**Comment by denesb** [comment]
New in v2:
* Use `get_table_if_exists()` instead of `try...catch`.

File: service/paxos/paxos_state.cc (hunk 1 above)
Comment (denesb) [reply service/paxos/paxos_state.cc]: Done in v2.

**Review by bhalevy — APPROVED** [review]

**Review by kostja — APPROVED** [review]

**Pull request closed without merging**